src/Renci.SshNet/Common/PipeOutputStream.cs

```diff
@@ -0,0 +1,111 @@
+﻿namespace Renci.SshNet.Common 
+{
+    using System;
+    using System.IO;
+
+    internal class PipeOutputStream : Stream
+    {
+        private LinkedListQueue<byte[]> _queue;
+        private bool _isDisposed;
+
+        public PipeOutputStream(LinkedListQueue<byte[]> queue)
+        {
+            _queue = queue;
+        }
+
+        public override void Flush()
+        {
+        }
+
+        public override long Seek(long offset, SeekOrigin origin)
+        {
+            throw new NotSupportedException();
+        }
+
+        public override void SetLength(long value)
+        {
+            throw new NotSupportedException();
```

Comment:
If we want to implement this, then the PipeOutputStream needs to know the current position of the PipeInputStream in to reject reducing the length to a value less than the current position.